libraries/core_libs/consensus/src/dag/dag_block_manager.cpp

```diff
@@ -263,12 +263,13 @@ std::pair<uint64_t, bool> DagBlockManager::getProposalPeriod(level_t level) {
   return result;
 }
 
-std::shared_ptr<ProposalPeriodDagLevelsMap> DagBlockManager::newProposePeriodDagLevelsMap(level_t anchor_level) {
+std::shared_ptr<ProposalPeriodDagLevelsMap> DagBlockManager::newProposePeriodDagLevelsMap(level_t anchor_level,
+                                                                                          uint64_t propose_period) {
   bytes period_levels_bytes = db_->getProposalPeriodDagLevelsMap(current_max_proposal_period_);
   assert(!period_levels_bytes.empty());
   ProposalPeriodDagLevelsMap last_period_levels_map(period_levels_bytes);
 
-  auto propose_period = ++current_max_proposal_period_;
+  current_max_proposal_period_ = propose_period;
```

Comment:
Since the propose period has to be continuity saving in DB, that will use in getProposalPeriod.
How to guaranty the propose period number continuity here?